### PR TITLE
Triggers refactor

### PR DESCRIPTION
**Pull request by lann**
Highlights:

- Rewrite Trigger trait as TriggerExecutor to support deduplication of
  CLI logic
- Add TriggerExecutorCommand and TriggerExecutorBuilder to deduplicate
  trigger command logic
- Deduplicate FollowComponents logic into ExecutionContextConfiguration
- Opportunistically refactor HttpTrigger:
  - Reduce duplication between `serve` & `serve_tls`
  - Replace `clone`ing of HttpTrigger with Arc wrapper

### CODE REVIEW

**Review by fibonacci1729 — APPROVED** [review]

File: crates/http/Cargo.toml [hunk 1]
```diff
@@ -13,7 +13,7 @@ async-trait = "0.1"
 atty = "0.2"
 bytes = "1.1"
 cap-std = "0.24.1"
-ctrlc = "3.2.1"
+clap = "3"
```
[the remark this answers is not in the document]
Comment (lann) [reply crates/http/Cargo.toml:16]: https://github.com/fermyon/spin/pull/513/files#diff-c6fae8d2510716579c8de5371b0212b266a1633f45e515fde66fc45f6d1dbf21R58

File: crates/trigger/src/lib.rs [hunk 2]
```diff
@@ -1,118 +1,124 @@
...
-    type Config;
-    /// component configuration
-    type ComponentConfig;
-    /// runtime configuration
-    type ExecutionConfig;
+pub trait TriggerExecutor: Sized {
+    type GlobalConfig;
+    type TriggerConfig;
+    type RunConfig;
+    type RuntimeContext: Default + 'static;
```
Comment (lann) [inline crates/trigger/src/lib.rs:16]: This renaming is aiming towards a future manifest rewrite a la https://hackmd.io/@lann/SyfiEmEHc

**Review by radu-matei — APPROVED** [review]
This is excellent!
I think this just needs to have the conflicts resolved, and it can be merged.

LGTM

**Comment by lann** [comment]
Unfortunately the conflicts are pretty significant. I'm going to try to deal with #520 first.

**Pull request merged**